libweb3jsonrpc/Eth.cpp

```diff
@@ -345,17 +345,28 @@ Json::Value Eth::eth_getBlockByHash(string const& _blockHash, bool _includeTrans
 {
 	try
 	{
-		h256 h = jsToFixed<32>(_blockHash);
+		h256 const h = jsToFixed<32>(_blockHash);
 		if (!client()->isKnown(h))
 			return Json::Value(Json::nullValue);
 
-		if (_includeTransactions)
-			return toJson(client()->blockInfo(h), client()->blockDetails(h), client()->uncleHashes(h), client()->transactions(h), client()->sealEngine());
-		else
-			return toJson(client()->blockInfo(h), client()->blockDetails(h), client()->uncleHashes(h), client()->transactionHashes(h), client()->sealEngine());
-	}
-	catch (...)
-	{
+        Json::Value ret;
+        auto const blockDetails = client()->blockDetails(h);
```

Comment:
Could you then also extract here the calls ro `blockInfo`, `uncleHashes`, `sealEngine`, not to duplicate them below